sources/packages/backend/apps/api/src/services/student/student-information.service.ts

```diff
@@ -31,6 +46,89 @@ export class StudentInformationService {
     });
   }
 
+  /**
+   * Get student applications.
+   * @param studentId student id.
+   * @returns applications.
+   */
+  async getStudentApplications(
+    studentId: number,
+  ): Promise<ApplicationDetail[]> {
+    // Get the SQL to get last 3 active program years including the current program year.
+    const programYearQuery = this.programYearRepo
+      .createQueryBuilder("programYear")
+      .select(["programYear.id"])
+      .where("programYear.active = true")
+      .andWhere("programYear.startDate <= NOW()")
+      .orderBy("programYear.startDate", "DESC")
+      .limit(3)
+      .getSql();
+
+    const queryResult = await this.applicationRepo
+      .createQueryBuilder("searchApplication")
+      .select([
+        "searchApplication.id",
+        "searchApplication.applicationNumber",
+        "searchApplication.applicationStatus",
+        "searchApplication.applicationStatusUpdatedOn",
+        "currentAssessment.id",
+        "currentAssessment.workflowData",
+        "currentAssessment.assessmentData",
+        "scholasticStanding.id",
+        "scholasticStanding.changeType",
+        "scholasticStanding.submittedData",
+        "offering.id",
+        "offering.studyBreaks",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "location.id",
+        "location.institutionCode",
+        "location.name",
+        "location.primaryContact",
+        "disbursementSchedule.id",
+        "disbursementSchedule.disbursementDate",
+        "disbursementSchedule.dateSent",
+        "disbursementSchedule.disbursementScheduleStatus",
+        "disbursementValue.id",
+        "disbursementValue.valueCode",
+        "disbursementValue.valueAmount",
+        "disbursementValue.effectiveAmount",
+      ])
+      .addSelect("searchApplication.data ->> 'dependants'", "dependants")
```

Comment:
The `searchApplication.data ->> 'dependants'` is already accessing dynamic data, with that in mind, is there a benefit in requesting the data from the DB twice instead of directly consuming the dependents from the `searchApplication.data`.
That said, do we really need the `ApplicationDetail` with the appended `dependants` array?

Reply:
With `searchApplication.data ->> 'dependants'` the `searchApplication.data` was removed to access only the dependants. 

To access just one property `dependants` I was trying to avoid retrieving the whole JSONB.